src/lib.rs

```diff
@@ -39,13 +42,30 @@ pub struct Scope {
     /// Imports
     imports: OrderMap<String, OrderMap<String, Import>>,
 
+    /// The modules defined in this scope.
+    /// 
+    /// Modules are stored in a map rather than in this scope's
+    /// `items` vector so that they may be accessed by name directly
+    /// after being created.
+    modules: OrderMap<RcKey, Module>,
```

Comment:
I would not bother with this, instead doing a scan of `items` to find a module. Again, lookup performance isn't really a factor at this point.

Reply:
Hmm, my initial implementation did a scan through `items`, but I ended up going with this approach because i found it to be simpler (esp. given the borrow-checker gymnastics that ended up being necessary to make `module_or_add` work).